src/matrix_input.jl

```diff
@@ -228,3 +229,81 @@ struct MILP{T, LP<:AbstractLPForm{T}}
     lp::LP
     variable_type::Vector{VariableType}
 end
+
+struct ConicForm{T, AT<:AbstractMatrix{T}} <: AbstractConicForm{T}
+    # assuming minimization for now
+    # direction::MOI.OptimizationSense 
+    c::Vector{T}
+    A::AT
+    b::Vector{T}
+    cones::Vector{<: MOI.AbstractVectorSet}
+end
+
+"""
+Convert a `MathOptInterface` model to `MatrixOptInterface`
+"""
+function getConicForm(model::M, con_idx) where {M <: MOI.AbstractOptimizer}
+    conic = ConicData()
+
+    # 1st allocate variables and constraints
+    N = MOI.get(model, MOI.NumberOfVariables())
+    __allocate_variables(conic, N)
+    for con in con_idx
+        func = MOI.get(model, MOI.ConstraintFunction(), con)
+        set = MOI.get(model, MOI.ConstraintSet(), con)
+        F = typeof(func)
+        S = typeof(set)
+        __allocate_constraint(conic, func, set)
+    end
+
+    __load_variables(conic, N)
+    
+    CONES_OFFSET = Dict(
+        MOI.Zeros => 0,
+        MOI.Nonnegatives => 0,
+        MOI.SecondOrderCone => 0,
+        MOI.PositiveSemidefiniteConeTriangle => 0,
+        MOI.ExponentialCone => 0, 
+        MOI.DualExponentialCone => 0
+    )
+
+    for con in con_idx
+        func = MOI.get(model, MOI.ConstraintFunction(), con)
+        set = MOI.get(model, MOI.ConstraintSet(), con)
+        F = typeof(func)
+        S = typeof(set)
+        __load_constraint(conic, CI{F, S}(CONES_OFFSET[S]), func, set)
+        CONES_OFFSET[S] += cons_offset(set)
+    end
+    
+    # now SCS data shud be allocated
+    A = sparse(
+        conic.data.I, 
+        conic.data.J, 
+        conic.data.V 
+    )
+    b = conic.data.b 
+
+    # extract `c`
+    obj = MOI.get(model, MOI.ObjectiveFunction{MOI.ScalarAffineFunction{Float64}}())
+    __load(conic, MOI.ObjectiveFunction{MOI.ScalarAffineFunction{Float64}}(), obj)
+    c = conic.data.c
+
+    # fix optimization sense
+    if MOI.get(model, MOI.ObjectiveSense()) == MOI.MAX_SENSE
+        c = -c
+    end
+
+    # reorder constraints
+    cis = sort(
+        con_idx, 
+        by = x->CONES[typeof(MOI.get(model, MOI.ConstraintSet(), x))]
+    )
+
+    # extract cones
+    cones = MOI.get(model, MOI.ConstraintSet(), cis)
+
+    return ConicForm{Float64, typeof(A)}(
```

Comment:
Could we specify `Float64` in the template of this function?

Reply:
added